File: App/src/main/java/com/fooddelivery/app/data/models/Order.kt
package com.fooddelivery.app.data.models

import com.google.gson.annotations.SerializedName

data class Order(
    @SerializedName("id")
    val id: Int,
    @SerializedName("buyer_id")
    val buyerId: Int,
    @SerializedName("seller_id")
    val sellerId: Int,
    @SerializedName("driver_id")
    val driverId: Int?,
    @SerializedName("order_number")
    val orderNumber: String,
    @SerializedName("status")
    val status: String,
    @SerializedName("subtotal")
    val subtotal: Double,
    @SerializedName("delivery_fee")
    val deliveryFee: Double,
    @SerializedName("total_amount")
    val totalAmount: Double,
    @SerializedName("delivery_address")
    val deliveryAddress: String,
    @SerializedName("delivery_phone")
    val deliveryPhone: String,
    @SerializedName("delivery_notes")
    val deliveryNotes: String?,
    @SerializedName("created_at")
    val createdAt: String,
    @SerializedName("updated_at")
    val updatedAt: String?,
    @SerializedName("delivered_at")
    val deliveredAt: String?,
    @SerializedName("items")
    val items: List<OrderItem>
)

data class OrderItem(
    @SerializedName("id")
    val id: Int,
    @SerializedName("food_id")
    val foodId: Int,
    @SerializedName("quantity")
    val quantity: Int,
    @SerializedName("unit_price")
    val unitPrice: Double,
    @SerializedName("subtotal")
    val subtotal: Double
)

data class OrderCreateRequest(
    @SerializedName("seller_id")
    val sellerId: Int,
    @SerializedName("items")
    val items: List<OrderItemCreate>,
    @SerializedName("delivery_address")
    val deliveryAddress: String,
    @SerializedName("delivery_phone")
    val deliveryPhone: String,
    @SerializedName("delivery_notes")
    val deliveryNotes: String? = null,
    @SerializedName("delivery_fee")
    val deliveryFee: Double = 0.0
)

data class OrderItemCreate(
    @SerializedName("food_id")
    val foodId: Int,
    @SerializedName("quantity")
    val quantity: Int
)

File: Backend-Api/schemas/order.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime
from models.order import OrderStatus


class OrderItemCreate(BaseModel):
    food_id: int
    quantity: int


class OrderItemResponse(BaseModel):
    id: int
    food_id: int
    quantity: int
    unit_price: float
    subtotal: float

    class Config:
        from_attributes = True


class OrderCreate(BaseModel):
    seller_id: int
    items: List[OrderItemCreate]
    delivery_address: str
    delivery_phone: str
    delivery_notes: Optional[str] = None
    delivery_fee: float = 0.0


class OrderResponse(BaseModel):
    id: int
    buyer_id: int
    seller_id: int
    driver_id: Optional[int]
    order_number: str
    status: OrderStatus
    subtotal: float
    delivery_fee: float
    total_amount: float
    delivery_address: str
    delivery_phone: str
    delivery_notes: Optional[str]
    created_at: datetime
    updated_at: Optional[datetime]
    delivered_at: Optional[datetime]
    items: List[OrderItemResponse]

    class Config:
        from_attributes = True


class OrderStatusUpdate(BaseModel):
    status: OrderStatus

